### PR TITLE
Remove `ReadonlyRepo::repo_path`

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
Looks like I messed up canonicalization. Will look into it later

**Comment by martinvonz** [comment]
> Looks like I messed up canonicalization. Will look into it later

That should be done now. I added canonicalization in `TestBackend`.

File: lib/src/repo.rs
Comment (yuja) [inline lib/src/repo.rs:1]: nit: typo in commit message: `Repoload::` -> `RepoLoader::`

File: lib/testutils/src/test_backend.rs
Comment (yuja) [inline lib/testutils/src/test_backend.rs:1]: nit: maybe better to add inline comment? It's subtle problem and uneasy to figure out why `.canonicalize()` is needed here.

File: lib/tests/test_git.rs [hunk 1]
```diff
@@ -159,14 +159,14 @@ fn test_import_refs() {
         view.get_remote_branch("main", "git"),
         &RemoteRef {
             target: RefTarget::normal(jj_id(&commit2)),
-            state: RemoteRefState::Tracking,
+            state: RemoteRefState::Tracking
```
Comment (yuja) [inline lib/tests/test_git.rs:162]: nit: unrelated formatting change in `test_git.rs`.

File: lib/testutils/src/lib.rs [hunk 2]
```diff
@@ -238,11 +245,13 @@ impl TestWorkspace {
             signer,
         )
         .unwrap();
+        let repo_path = workspace_root.join(".jj").join("repo");
```
Comment (yuja) [inline lib/testutils/src/lib.rs:248]: nit: might be better to use `workspace.workspace_root()` or just forward to `workspace.repo_path()`. `Workspace::init*()` does path canonicalization.

File: lib/src/workspace.rs [hunk 3]
```diff
@@ -150,24 +151,28 @@ fn init_working_copy(
...
     pub fn new(
         workspace_root: &Path,
+        repo_path: PathBuf,
         working_copy: Box<dyn WorkingCopy>,
         repo_loader: RepoLoader,
     ) -> Result<Workspace, PathError> {
         let workspace_root = workspace_root.canonicalize().context(workspace_root)?;
         Ok(Self::new_no_canonicalize(
             workspace_root,
+            repo_path,
```
Comment (yuja) [inline lib/src/workspace.rs:161]: Maybe it's safer to do `repo_path.canonicalize()` here? It was previously obtained from `repo.loader()`, which I think was canonicalized.

File: lib/tests/test_init.rs [hunk 4]
```diff
@@ -42,7 +42,6 @@ fn test_init_local() {
         .backend_impl()
         .downcast_ref::<GitBackend>()
         .is_none());
-    assert_eq!(repo.repo_path(), &canonical.join(".jj").join("repo"));
```
Comment (yuja) [inline lib/tests/test_init.rs:45]: nit: `workspace.repo_path()` could be tested instead

**Review by yuja — APPROVED** [review]

File: lib/src/repo.rs
Comment (martinvonz) [reply lib/src/repo.rs:1]: Done

File: lib/testutils/src/test_backend.rs
Comment (martinvonz) [reply lib/testutils/src/test_backend.rs:1]: Good idea. I added a comment on the map.

File: lib/src/workspace.rs (hunk 3 above)
Comment (yuja) [reply lib/src/workspace.rs:161]: Oh, but `load()`-ed workspace wasn't? That's odd.

File: lib/tests/test_git.rs (hunk 1 above)
Comment (martinvonz) [reply lib/tests/test_git.rs:162]: Fixed. Not sure how that happened.

File: lib/testutils/src/lib.rs (hunk 2 above)
Comment (martinvonz) [reply lib/testutils/src/lib.rs:248]: Ah, thanks for catching that. Fixed
